dropwizard-migrations/src/main/java/io/dropwizard/migrations/AbstractLiquibaseCommand.java

```diff
@@ -72,16 +75,21 @@ private CloseableLiquibase openLiquibase(final PooledDataSourceFactory dataSourc
             throws SQLException, LiquibaseException {
         final CloseableLiquibase liquibase;
         final ManagedDataSource dataSource = dataSourceFactory.build(new MetricRegistry(), "liquibase");
-
+        final Database database = createDatabase(dataSource, namespace);
         final String migrationsFile = namespace.getString("migrations-file");
         if (migrationsFile == null) {
-            liquibase = new CloseableLiquibaseWithClassPathMigrationsFile(dataSource, migrationsFileName) {
-            };
+            liquibase = new CloseableLiquibaseWithClassPathMigrationsFile(dataSource, database, migrationsFileName);
         } else {
-            liquibase = new CloseableLiquibaseWithFileSystemMigrationsFile(dataSource, migrationsFile);
+            liquibase = new CloseableLiquibaseWithFileSystemMigrationsFile(dataSource, database, migrationsFile);
         }
 
-        final Database database = liquibase.getDatabase();
+        return liquibase;
+    }
+
+    private Database createDatabase(final ManagedDataSource dataSource, Namespace namespace) throws SQLException, LiquibaseException {
```

Comment:
I think the `dataSource` variable could be declared without `final`, because it's not shared. From the other point of view, the `openLiquibase` method declares it parameter names as `final`, but it seems redundant.